### DISCUSSION TITLE
Performance Effect of SFBO implementation

### DISCUSSION THREAD

**Discussion by omerguzelelectronicguy** [riscv-boom/riscv-boom#685] [Q&A] [answered]
Hello, I want to ask about sfbo implementation effect.
> As an example, SonicBOOM achieves 6.15 CoreMark/MHz with the SFB optimization enabled, compared to 4.9 CoreMark/MHz without.

It is taken from [SonicBOOM: The 3rd Generation Berkeley Out-of-Order Machine](https://carrv.github.io/2020/papers/CARRV2020_paper_15_Zhao.pdf)

My question is did you make any compiler modification to get more benefit from sfbo optimization?
If it's the case, did you get 4.9 with not modified compiler and 6.15 with modified compiler? or you get both with modified compiler? 

Thank you so much.

**Comment by jerryz123** [comment, marked as answer]
The optimization for short forwards branches was already implemented in upstream GCC, we only had to enable it.

**Reply by omerguzelelectronicguy** [reply]
Ok, I thought there should be an implementation-specific optimization like putting the SFBO margin to a 128-bit cache line or so. So you didn't make any additional backend improvement in the compiler for sfbo optimization.

According to my second question, Did you get both points with the compiled code that the SFBO flag enabled? or the higher point is the SFBO flag enabled with hardware support and the lower one is the SFBO flag disabled with no hardware support?

Thank you so much.
